whisper/whisperv5/api.go

```diff
@@ -69,20 +69,24 @@ func NewPublicWhisperAPI(w *Whisper) *PublicWhisperAPI {
 // this loop deletes filter that have not been used within filterTimeout
 func (api *PublicWhisperAPI) run() {
 	timeout := time.NewTicker(2 * time.Minute)
+	defer timeout.Stop()
 	for {
-		<-timeout.C
-
-		api.mu.Lock()
-		for id, lastUsed := range api.lastUsed {
-			if time.Since(lastUsed).Seconds() >= filterTimeout {
-				delete(api.lastUsed, id)
-				if err := api.w.Unsubscribe(id); err != nil {
-					log.Error("could not unsubscribe whisper filter", "error", err)
+		select {
+		case <-api.w.quit:
+			return
```

Comment:
I'm nitpicking here: I would rather have a `break` than a `return` in case someone in the future adds some cleanup code and misses out.

Reply:
@dshulyak please disregard my previous comment, I just realized that this is going to disappear in v6 so future maintainability is not a concern.